### ISSUE TITLE
Missing Helm release secrets cause ListHelmReleases to return an error, and to list no releases

### ISSUE THREAD

**Issue by devmage** [weaveworks/weave-gitops#2664]
Hey friends! @amulyakish and I encountered what we're pretty sure is a corner-case in WGO's rendering of deployed HelmReleases.

The behavior is similar to what was seen in #2016 and a few other issues. The associated fix for #2016 , #2042 , introduced behavior that requires all Helm releases in `ListHelmReleases` to be "complete", i.e. to have a Helm [release secret](https://codeengineered.com/blog/2020/helm-secret-storage/) created for a given release revision. Without the Helm release secret, WGO is not able to read the details of the release. Without the details of the release, WGO does not render _any_ HelmReleases, renders a stack trace instead, and yields an error about a missing Helm secret.

**Describe the bug**

If a Flux HelmRelease receives a new Helm revision number, but the associated Helm release at that revision did not complete and did not create a release secret, then there is no release to read. WGO therefore does not render the *entire list* of HelmReleases. We also see a stack trace, and a console message like `"secrets \"sh.helm.release.v1.myapp8v9np.v1\" not found"`.

**Environment**

- Weave-Gitops Version: 0.9.3 and `main` `HEAD`
- Flux Version: v0.28.5
- Kubernetes version: 1.22

**To Reproduce**

Introduce a Flux HelmRelease with a Helm revision number that doesn't exist. This may be due to a Helm upgrade that is attempted, but that fails due to Kubernetes validation rules.

We saw an example of this in our own cluster. Following one cluster's upgrade to Kubernetes 1.22, the Ingress object at `extensions/v1beta1` was [deprecated](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#ingress-v122). One application had not updated their Helm chart for this, causing a Helm upgrade failure. Some sample outputs:

```
$ kubectl get hr -n this-namespace
NAME                     AGE    READY   STATUS
this-service   355d   False   Helm upgrade failed: current release manifest contains removed kubernetes api(s) for this kubernetes version and it is therefore unable to build the kubernetes objects for performing the diff. error from kubernetes: unable to recognize "": no matches for kind "Ingress" in version "extensions/v1beta1"
```

```yaml
$ kubectl describe hr -n this-namespace this-service
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  creationTimestamp: "2021-09-08T01:29:15Z"
  finalizers:
  - finalizers.fluxcd.io
  generation: 50
  labels:
    kustomize.toolkit.fluxcd.io/name: applications
    kustomize.toolkit.fluxcd.io/namespace: internal-system
  managedFields:
  ...
spec:
  ...
status:
  conditions:
  - lastTransitionTime: "2022-08-25T21:22:42Z"
    message: 'Helm upgrade failed: current release manifest contains removed kubernetes
      api(s) for this kubernetes version and it is therefore unable to build the kubernetes
      objects for performing the diff. error from kubernetes: unable to recognize
      "": no matches for kind "Ingress" in version "extensions/v1beta1"'
    reason: UpgradeFailed
    status: "False"
    type: Ready
  - lastTransitionTime: "2022-08-25T21:22:42Z"
    message: |-
      Helm upgrade failed: current release manifest contains removed kubernetes api(s) for this kubernetes version and it is therefore unable to build the kubernetes objects for performing the diff. error from kubernetes: unable to recognize "": no matches for kind "Ingress" in version "extensions/v1beta1"

      Last Helm logs:

      preparing upgrade for this-service
      resetting values to the chart's original version
      performing update for this-service
    reason: UpgradeFailed
    status: "False"
    type: Released
  failures: 1
  ...
  lastReleaseRevision: 35 ## <----- NOTICE HERE!
  observedGeneration: 50
  upgradeFailures: 1
```

Compare the above `lastReleaseRevision` to the following `REVISION`:

```
$ helm list -n this-namespace
NAME          NAMESPACE                 REVISION        UPDATED                                      STATUS          CHART                    APP VERSION
this-service  this-namespace            34              2022-07-21 16:29:31.442591761 +0000 UTC      deployed        this-service-1.0.0       1.0.0
...
```
**Expected behavior**

WGO Applications lists all HelmReleases, regardless of deployment state.

**Actual Behavior**

WGO Applications lists *no* HelmReleases, regardless of deployment state. WGO instead shows an error message.

![image](https://user-images.githubusercontent.com/1923665/187300360-e9753626-232b-4870-9a9e-fd07874aac2f.png)

**Additional Context (screenshots, logs, etc)**

What might we do about this? In the short term, there is at least one relatively simple fix, and I'll have a PR open shortly to show. The logic in `ListHelmReleases` may be amended to append `err` to `respErrors` on failure to `getHelmReleaseInventory`, instead of returning early: https://github.com/weaveworks/weave-gitops/blob/main/core/server/helm_release.go#L58-L67

This will cause the Helm Release listing in WGO to *omit* the failing release in its list view. Not great, but it permits the Helm Release list view to render overall in the case identified in this issue, thereby satisfying this issue. Users will not see entries that do not have releases at their specified revision. *This is confusing* for users, and users will need to a) notice that their release is missing, and then b) manually dig into `kubectl` and `helm` in their clusters to figure out what happened to their missing entry. It is however arguably better than one failing release causing WGO to not render _any_ releases, which is the current state.

In the long-term, however, I'm not sure what you all might want to do with error handling and messaging, and I don't mean to presume at this stage. The inner-loop error handling logic in most of the other List methods looked fine for now. I saw a couple of spots in both List and Get methods that seemed like they would push stack traces back to the user (`return nil, err` being the indicator). All of those are out of scope of this Issue.

There is at least one other List endpoint that might warrant a similar review as here, in `ListKustomizations`: https://github.com/weaveworks/weave-gitops/blob/main/core/server/kustomization.go#L66-L69

Thank you for your consideration here! PR for `ListHelmReleases` coming shortly.

**Comment by devmage** [comment]
^ Just to put context to this close-reopen dance, #2665 was a stop-gap measure as mentioned above. It isn't a fully baked solution to this issue.

**Comment by ozamosi** [comment]
This was fixed by #2765 - helmreleases now show up even if they don't have a secret.
